create_jenkins_job.py

```diff
@@ -127,6 +129,25 @@ def create_job(os_name, job_name, template_file, additional_dict):
         job_config = expand_template(template_file, job_data)
         configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
 
+    def create_linux_based_nightly_jobs():
+        # configure nightly job for compiling with clang on linux
+        create_job('linux', 'nightly_linux_clang', 'ci_job.xml.em', {
```

Comment:
So I think this can be applied to more than just Linux, and, specifically, more than just clang.

Perhaps we could add a GCC address sanitizer build as well? I think MSVC supports it as well, but I'd need to double check that statement.

I really like this idea, by the way. Sanitizers are super useful.

Reply:
Thanks Allen. Currently I have added a GCC address sanitizer on linux. I will be adding thread sanitizer as well soon. I am not adding any clang sanitizers for now.